ansible_rulebook/rule_set_runner.py

```diff
@@ -435,6 +447,46 @@ async def _call_action(
             except BaseException as e:
                 logger.error(e)
                 raise
+        elif action_plugin := self.find_action(action):
```

Comment:
@benthomasson are we going to have special names for the custom actions so they don't conflict with our builtin action names. If the name conflict exists we will only run the builtin and not run the custom actions.

Reply:
@benthomasson Since the code is the same for the 2 find_actions call can we move it into a separate function
```
         def run_action_plugin(self,....):
                action_plugin = self.find_action(action) or find_action(*split_collection_name(action))
                if not action_plugin:
                   return
                    
          try:
                await action_plugin["main"](
                    event_log=self.event_log,
                    inventory=inventory,
                    hosts=hosts,
                    variables=variables,
                    project_data_file=self.project_data_file,
                    source_ruleset_name=ruleset,
                    source_ruleset_uuid=ruleset_uuid,
                    source_rule_name=rule,
                    source_rule_uuid=rule_uuid,
                    rule_run_at=rule_run_at,
                    **action_args,
                )
            except Exception as e:
                logger.error("Error calling action %s, err %s", action, str(e))
            except BaseException as e:
                logger.error(e)
                raise
```

Reply:
@benthomasson doing this search with every action call might not make sense, we could load all the actions in when we read the ruleset. Which also allows us to ensure that all the actions are valid before we start accepting the events and running the rule engine. if the actions are missing we can do an early termination to indicate that actions are missing.

Reply:
> @benthomasson are we going to have special names for the custom actions so they don't conflict with our builtin action names. If the name conflict exists we will only run the builtin and not run the custom actions.

We should support FQCN for actions.   Names without collections prefixes are either builtin or loaded from a local directory.  The precedence should be builtin and then local directory for non-FQCN.